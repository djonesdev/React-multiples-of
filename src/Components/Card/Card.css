.c-card {
  border: 1px solid black;
  border-radius: 5px;
  margin: 5px;
  flex-basis: 30%;
}

@media (min-width: 768px) and (max-width: 1024px) {
  .c-card {
    flex-basis: 45%;
  }
}

@media only screen and (max-width: 768px) {
  .c-card {
    flex-basis: 90%;
  }
}
