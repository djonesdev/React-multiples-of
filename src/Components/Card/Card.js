import React from 'react'

import "./Card.css"

function Card({ children, onClick, classNames }) {
    return (
        <div onKeyPress={onClick} role="button" className={`c-card ${classNames}`} tabindex="0" onClick={onClick}>
            {children}
        </div>
    )
}

export default Card
