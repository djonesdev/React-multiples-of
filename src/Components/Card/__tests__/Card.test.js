import React from "react";
import renderer from "react-test-renderer";
import Card from "../Card";

test("Card renders correctly with callback props", () => {
    const mockFunc = jest.fn()
  const component = renderer.create(
    <Card onClick={mockFunc} page="http://www.facebook.com">
      <p>Some Test Content</p>
    </Card>
  );

  //Checks render
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Checks callback was called
  tree.props.onClick();
  expect(mockFunc).toBeCalledTimes(1)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
