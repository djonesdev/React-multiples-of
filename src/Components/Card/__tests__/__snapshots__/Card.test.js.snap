// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly with callback props 1`] = `
<div
  className="c-card undefined"
  onClick={[MockFunction]}
  onKeyPress={[MockFunction]}
  role="button"
  tabindex="0"
>
  <p>
    Some Test Content
  </p>
</div>
`;

exports[`Card renders correctly with callback props 2`] = `
<div
  className="c-card undefined"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onKeyPress={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  role="button"
  tabindex="0"
>
  <p>
    Some Test Content
  </p>
</div>
`;
